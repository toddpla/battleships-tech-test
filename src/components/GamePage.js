import React from 'react'
import { connect } from 'react-redux';
import { nextPlayer } from '../actions/GameActions'
import Grid from './Grid'


export class GamePage extends React.Component {

  render() {
    return (
      <div>
        <h1>Current Player: {this.props.current_player.name}</h1>
        <button onClick={this.props.nextPlayer} > Next Player</button>
        <Grid grid={this.props.game.grid}/>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  game: state.game,
  current_player: state.game[state.game.current_player],
})

const mapDispatchToProps = (dispatch) => ({
  nextPlayer: () => dispatch(nextPlayer())
})

export default connect(mapStateToProps, mapDispatchToProps)(GamePage)
