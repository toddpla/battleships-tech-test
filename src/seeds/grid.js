
export default new Array(100).fill(undefined)
