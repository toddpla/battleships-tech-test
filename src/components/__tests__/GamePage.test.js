import React from 'react'
import { GamePage } from '../GamePage'
import renderer from 'react-test-renderer';
import players from '../../fixtures/players'
import grid from '../../seeds/grid'

let game = {
  grid
}

test('renders correctly', () => {
  const setupScreen = renderer.create(<GamePage game={game} current_player={players[0]} />).toJSON();
  expect(setupScreen).toMatchSnapshot();
});
