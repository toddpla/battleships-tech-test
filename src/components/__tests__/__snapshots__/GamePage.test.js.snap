// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Current Player: 
    Sophie
  </h1>
  <button>
     Next Player
  </button>
  <div
    className="ship-list"
  >
    <h1>
      Ships
    </h1>
    <article>
      <div
        className="ship false"
        id="ship-Carrier"
        onClick={[Function]}
      >
        Carrier
      </div>
      <div
        className="ship false"
        id="ship-Battleship"
        onClick={[Function]}
      >
        Battleship
      </div>
      <div
        className="ship false"
        id="ship-Cruiser"
        onClick={[Function]}
      >
        Cruiser
      </div>
      <div
        className="ship false"
        id="ship-Submarine"
        onClick={[Function]}
      >
        Submarine
      </div>
      <div
        className="ship false"
        id="ship-Destroyer"
        onClick={[Function]}
      >
        Destroyer
      </div>
    </article>
  </div>
</div>
`;
